### DISCUSSION TITLE
live audio stream analysis

### DISCUSSION THREAD

**Discussion by unosonic** [tphakala/birdnet-go#32] [Show and tell]
sometimes I put a little streambox into a park, forest or similar spaces. It's raspberry pi zero based for best battery life, so no option to run birdnet-pi or birdnet-go. The box sends an mp3 stream to an icecast streaming server, nothing special here. I've thought to do the
birdnet-go analysis server-side, but since only rtsp is or will be supported, I've made a small script w/ ffmpeg and inotify:

```
#!/bin/bash
# grab http stream STREAM in background, create 15sec long pieces, convert them to 16bit mono wavs in DIR as outx.wav, x: 001, 002,...
ffmpeg -v quiet -i STREAM -f segment -segment_time 15 -acodec pcm_s16le -ac 1 -ar 16000 DIR/out%03d.wav &
# watch for files in folder DIR and analyze them when written, i.e. close_write event occured, delete file when done  
while segment=$(inotifywait --quiet --event close_write --format %w%f DIR/); do
        ./birdnet-go file $segment
        rm -f $segment
done
```
of course, it would be more convienient if birdnet-go would support also HTTP-based streams a input... ;)

**Comment by Gerwin12** [comment]
Thank you for this post. This was exactly what I was thinking off to reduce power consumption offgrid. However, I have no experience with icecast, as far as I understand it should run on the raspberry (the little streambox)??? Do you use 4G/LTE to connect to the internet? How much data will be streamed in mp3, what bitrate do you use, how to avoid port forwarding? Sorry for the many questions. best regards, Gerwin
